tensorflow/compiler/xla/service/gpu/conv_layout_normalization.cc

```diff
@@ -133,10 +137,19 @@ StatusOr<HloInstruction*> UpdateLayoutForCudnnConvolution(
       new_op = normalized_op;
     } else {
       new_op = MakeBitcastHlo(op, s_reordered);
+      performed_normalization = true;
     }
     normalized_operands.push_back(new_op);
   }
 
+  // Avoid replacing the Custom Call with an identical copy.
+  if (!performed_normalization &&
+      ShapeUtil::Equal(normalized_shape, hlo->shape()) &&
+      ConvolutionDimensionNumbersToString(new_dim_numbers) ==
+          ConvolutionDimensionNumbersToString(dim_numbers)) {
+    return std::nullopt;
+  }
```

Comment:
Why avoid replacing the custom call with an identical copy, when we didn't have this logic before?

Reply:
When the Custom Call returns a tuple, replacing the instruction with an identical copy can cause layout normalization to run in an endless loop.